hazelcast/src/hazelcast/client/protocol.cpp

```diff
@@ -547,15 +559,15 @@ namespace hazelcast {
             bool ClientMessageBuilder::appendExistingPartialMessage(std::unique_ptr<ClientMessage> &msg) {
                 bool result = false;
 
-/* TODO
-                MessageMap::iterator foundItemIter = partialMessages.find(msg->getCorrelationId());
-                if (partialMessages.end() != foundItemIter) {
-                    foundItemIter->second->append(msg.get());
-                    if (msg->isFlagSet(ClientMessage::END_FLAG)) {
+                auto found = partialMessages.find(msg->getCorrelationId());
```

Comment:
Do the intermediate and the end message fragments have the correlationId? I'm looking at the protocol doc, and as far as I understand, only the first frame of the message contain the correlationId.
Shouldn't we use the fragment id to match fragments of the same message?